From 95c39116b52c14c88f5b76f5cd8d56af995d7720 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Thu, 22 Dec 2016 22:04:42 +0100
Subject: [PATCH] init.rc: don't start console unless asked to

This can have a major impact on performance.

We don't want this, even on userdebug/eng builds.

Use the new property "ro.console.enable" to
enable the console service explicitly.

Change-Id: I93e7c65e92261443d1c9c70cfef9aa2ed5ff328a
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 43258782c..4b04b9e74 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1318,7 +1318,7 @@ on property:ro.debuggable=1
     # Give reads to anyone for the accessibility trace folder on debug builds.
     chmod 0775 /data/misc/a11ytrace
 
-on init && property:ro.debuggable=1
+on init && property:ro.debuggable=1 && property:ro.console.enable=1
     start console
 
 on userspace-reboot-requested
-- 
2.43.0

